### ISSUE TITLE
Vertebral labelling - Deep learning 

### ISSUE THREAD

**Issue by lrouhier** [spinalcordtoolbox/spinalcordtoolbox#2438]
- Extract metrics from the spine generic dataset to determine anatomical constraint that can be apply to vertebral labelling  : 

  * Extract coordinate of the label of each disc: XXX_labeled_discs.nii.gz 
  * Extract  the centerline coordinate of the spinal cord at each disc level (to get from segmentation ) (file : XXX_centerline.nii.gz )
  * Perform a statistical analysis of the inter-disc and disc-spinal cord distance for each disc.  

- develop and implement a pre-processing program to extract the median slice of an MRI and normalize it and modify it (with contrast enhancing filter) to integrate to a PyTorch loader

- Write existing cod in PyTorch syntax to integrate to sct and allow an easier use of custom loss  :

   * Implement a loader for training
   * Implement a previously existing network in PyTorch 

- Add constraint and evaluate the performance

    * implement custom loss function using the previously extracted metrics and 

    * Evaluate performance 

- SCT integration
    * implement an algorithm to generate a 3D nifti picture based on the coordinate obtained as output of the neural network 

    * combining previously implemented program as a pipeline and test it. 
 

REF
Ullmann et al. 
filesystem:chrome-extension://bomfdkbfpdhijjbeoicnfhjbdhncfhig/persistent/F3E9AB52871A11E58404CE482B0700DC/pp_attachments/c6b5f2fa-bdb2-069b-a9ed-3e5785c06204

**Comment by joshuacwnewton** [comment]
Closing, as this issue has been superseded by https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3793.
